File: Webgui/web/js/model/opmerking.js
define(['knockout'],
	function (ko) {

	return function(){
		var _this = this;

		_this.id = ko.observable();
		_this.identificatie = ko.observable();
		_this.opmerking = ko.observable();
		_this.tijd = ko.observable();
		_this.medewerker = ko.observable();
		_this.medewerkerId = ko.observable();
		_this.soortEntiteit = ko.observable();
		_this.entiteitId = ko.observable();
		_this.parentIdentificatie = ko.observable();

		_this.idIsNull = ko.computed(function(){
		    return _this.identificatie() == null;
		})
		_this.idIsNietNull = ko.computed(function(){
		    return _this.identificatie() != null;
		})
    };
});

File: Commons/src/main/java/nl/lakedigital/djfc/commons/domain/Opmerking.java
package nl.lakedigital.djfc.commons.domain;

import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlType;

@XmlRootElement(name = "opmerking")
@XmlType(namespace = "commons.domain")
public class Opmerking extends AbstracteEntiteitMetSoortEnId {
    private String tijd;
    private String opmerking;
    private String medewerker;
    private Long medewerkerId;

    public Opmerking() {
    }

    public Opmerking(SoortEntiteit soortEntiteit, Long entiteitId, String identificatie, String tijd, String opmerking, String medewerker, Long medewerkerId) {
        super(soortEntiteit, entiteitId, identificatie);
        this.tijd = tijd;
        this.opmerking = opmerking;
        this.medewerker = medewerker;
        this.medewerkerId = medewerkerId;
    }

    public String getTijd() {
        return tijd;
    }

    public void setTijd(String tijd) {
        this.tijd = tijd;
    }

    public String getOpmerking() {
        return opmerking;
    }

    public void setOpmerking(String opmerking) {
        this.opmerking = opmerking;
    }

    public String getMedewerker() {
        return medewerker;
    }

    public void setMedewerker(String medewerker) {
        this.medewerker = medewerker;
    }

    public Long getMedewerkerId() {
        return medewerkerId;
    }

    public void setMedewerkerId(Long medewerkerId) {
        this.medewerkerId = medewerkerId;
    }
}

